widgets/splitbutton/org.eclipse.nebula.widgets.splitbutton/src/org/eclipse/nebula/widgets/splitbutton/SplitButton.java

```diff
@@ -179,30 +179,26 @@ private void drawTriangle() {
 	}
 
 	private void addMouseDownListener() {
-		super.addListener(SWT.MouseDown, new Listener() {
-
-			@Override
-			public void handleEvent(final Event event) {
-				if (shouldShowMenu(event.x, event.y)) {
-					final Button button = (Button) event.widget;
-					final Rectangle rect = button.getBounds();
-					final Point p = button.toDisplay(rect.x, rect.y + rect.height);
-					getMenu().setLocation(p.x - rect.x, p.y - rect.y);
-					getMenu().setVisible(true);
-
-				} else {
-					for (final Listener listener : listeners) {
-						final Event evt = new Event();
-						evt.widget = SplitButton.this;
-						evt.display = getDisplay();
-						evt.type = SWT.Selection;
-						listener.handleEvent(evt);
-					}
+		super.addListener(SWT.MouseDown, event -> {
+			boolean shouldShowMenu = event.x >= splitButtonAreaLeft && event.y >= splitButtonAreaTop

```

Comment:
For readability the shouldShowMenu method should not be inlined.

Reply:
Ok, just reverted that part.